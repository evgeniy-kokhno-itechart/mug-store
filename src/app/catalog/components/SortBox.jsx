import React from 'react';
import { PropTypes } from 'prop-types';
import { DropdownInline } from '../../shared';

const SortBox = ({ sortColumnKey, sortOptions, onSortChange }) => (
  <DropdownInline
    name="sortDropdown"
    label="Sorted by:"
    options={sortOptions}
    value={sortColumnKey}
    wrapperClasses="justify-content-end mt-1 mt-lg-0"
    selectClasses='form-control'
    onChange={onSortChange}
  />
);

SortBox.propTypes = {
  sortOptions: PropTypes.arrayOf(PropTypes.object).isRequired,
  sortColumnKey: PropTypes.string.isRequired,
  onSortChange: PropTypes.func.isRequired,
};

export default SortBox;
