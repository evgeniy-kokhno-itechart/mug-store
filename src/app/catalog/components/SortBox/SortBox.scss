@import "Variables";

.sort-box {
    display: flex;
    justify-content: flex-end;

    &__dropdown {
        padding: $padding-vertical;
    }
}