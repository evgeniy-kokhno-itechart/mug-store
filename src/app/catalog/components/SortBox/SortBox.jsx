import React from 'react';
import { PropTypes } from 'prop-types';
import { DropdownInline } from '../../../shared';
import './SortBox.scss';

const SortBox = ({
  sortColumnKey, sortOptions, customClasses, onSortChange,
}) => (
  <DropdownInline
    name="sortDropdown"
    label="Sorted by"
    options={sortOptions}
    value={sortColumnKey}
    wrapperClasses={`sort-box ${customClasses}`}
    selectClasses='sort-box__dropdown'
    onChange={onSortChange}
  />
);

SortBox.propTypes = {
  sortOptions: PropTypes.arrayOf(PropTypes.object).isRequired,
  sortColumnKey: PropTypes.string.isRequired,
  onSortChange: PropTypes.func.isRequired,
  customClasses: PropTypes.string,
};

SortBox.defaultProps = {
  customClasses: '',
};

export default SortBox;
