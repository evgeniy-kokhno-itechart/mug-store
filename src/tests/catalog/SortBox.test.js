/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SortBox from '../../catalog/components/SortBox';

configure({ adapter: new Adapter() });

describe('<SortBox />', () => {
  let SortBoxWrapper;

  beforeEach(() => {
    const onSortChange = jest.fn();
    SortBoxWrapper = shallow(
      <SortBox sortColumnKey="title" sortOptions={[{ id: 'title_desc', name: 'Title Z-A' }]} onSortChange={onSortChange} />,
    );
  });

  it('check render properly', () => {
    expect(SortBoxWrapper.debug()).toMatchSnapshot();
  });
});
