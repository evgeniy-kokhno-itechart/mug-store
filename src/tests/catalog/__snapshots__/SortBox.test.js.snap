// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortBox /> check render properly 1`] = `
<DropdownInline
  defaultText=""
  error=""
  label="Sorted by"
  labelClasses=""
  name="sortDropdown"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "title_desc",
        "name": "Title Z-A",
      },
    ]
  }
  selectClasses="sort-box__dropdown"
  value="title"
  wrapperClasses="sort-box "
/>
`;
